wagtail/images/models.py

```diff
@@ -412,33 +433,112 @@ def get_rendition(self, filter: Union["Filter", str]) -> "AbstractRendition":
         Note: If using custom image models, an instance of the custom rendition
         model will be returned.
         """
-        if isinstance(filter, str):
-            filter = Filter(spec=filter)
+        if isinstance(filter, Filter):
+            filter_spec = filter.spec
+        else:
+            filter_spec = filter
+        return self.get_renditions(filter_spec)[filter_spec]
+
+    def get_renditions(self, *filter_specs: str) -> Dict[str, "AbstractRendition"]:
+        """
+        Returns a ``dict`` of ``Rendition`` instances with image files reflecting
+        the supplied ``filter_specs``, keyed by the relevant ``filter_spec`` string.
 
+        Note: If using custom image models, instances of the custom rendition
+        model will be returned.
+        """
         Rendition = self.get_rendition_model()
+        filter_map: Dict[str, Filter] = {
+            spec: Filter(spec) for spec in set(filter_specs)
+        }
+        unfulfilled: List[Filter] = list(filter_map.values())
+        return_value: Dict[str, AbstractRendition] = {}
 
         try:
-            rendition = self.find_existing_rendition(filter)
-        except Rendition.DoesNotExist:
-            rendition = self.create_rendition(filter)
-            # Reuse this rendition if requested again from this object
-            if "renditions" in getattr(self, "_prefetched_objects_cache", {}):
-                self._prefetched_objects_cache["renditions"]._result_cache.append(
-                    rendition
-                )
-            elif hasattr(self, "prefetched_renditions"):
-                self.prefetched_renditions.append(rendition)
+            renditions_cache = caches["renditions"]
+        except InvalidCacheBackendError:
+            renditions_cache = None
 
-        try:
-            cache = caches["renditions"]
-            key = Rendition.construct_cache_key(
-                self.id, filter.get_cache_key(self), filter.spec
+        # Renditions to be added to the cache (when rendition caching is enabled)
+        cache_additions: List[Tuple[Filter, AbstractRendition]] = []
+
+        # Interrogate prefetched values first (if available)
+        prefetched = self._get_prefetched_renditions()
+        if prefetched is not None:
+            # NOTE: When renditions are prefetched, it is assumed that if
+            # requested renditions exist, they will be present, and further
+            # cache/database lookups are avoided.
+
+            for spec, filter in filter_map.items():
+                cache_key = filter.get_cache_key(self)
+                for rendition in prefetched:
```

Comment:
Suggestion: It might be better to build a map of `spec: rendition` based on the prefetched before we enter this loop, to avoid the need for an inner loop.

Reply:
Have split this into two loops in a26891e, so that at this point in the code, we're only looping through (and generating cache keys for) specs that we have potential matches for.